staging/src/k8s.io/apiserver/pkg/endpoints/handlers/update.go

```diff
@@ -110,12 +111,28 @@ func UpdateResource(r rest.Updater, scope RequestScope, admit admission.Interfac
 		admit = admission.WithAudit(admit, ae)
 
 		if err := checkName(obj, name, namespace, scope.Namer); err != nil {
+			err = errors.NewInternalError(err)
 			scope.err(err, w, req)
 			return
 		}
 
 		userInfo, _ := request.UserFrom(ctx)
-		var transformers []rest.TransformFunc
+		transformers := []rest.TransformFunc{}
+		if utilfeature.DefaultFeatureGate.Enabled(features.ServerSideApply) {
+			if scope.OpenAPIModels == nil {
+				fmt.Println("No OpenAPI for: %v", scope.Kind)
+			} else {
+				converter, err := apply.NewTypeConverter(scope.OpenAPIModels)
```

Comment:
It'd be nice if we didn't have to create the type converter on every request. Can it be updated during the lifetime of the apiserver @jennybuckley ?

Reply:
Thought about this as well. Same for the VersionConverter?

Reply:
the VersionConverter is almost free to build, so I don't care as much

Reply:
would it be a problem at all if we had separate processes concurrently using the same typeConverter?

Reply:
I don't think so. As long as we ensure it is there when concurrent processes use it, this should be fine.
Where would it be best to store then @jennybuckley ?